ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -317,11 +363,22 @@ boolean isCaughtUp() {
   void shutdown() throws InterruptedException, StoreException, IOException {
     if (running.get()) {
       running.set(false);
-      synchronized (lock) {
-        lock.notify();
+      boolean isPaused = isPaused();
+      if (isPaused) {
+        reentrantLock.lock();
+        paused.set(false);
+        pauseCondition.signal();
+        reentrantLock.unlock();
+      } else {
+        synchronized (reentrantLock) {
```

Comment:
this is very confusing (using the lock variable as an object for locking) - i think you should use the pauseCondition for both the timed wait (sleep time) and the actual pause.